launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -136,7 +136,8 @@ void addOptionString(List<String> cmd, String options) {
 
     Set<String> cp = new LinkedHashSet<>();
     addToClassPath(cp, appClassPath);
-
+    addToClassPath(cp, getenv("SPARK_DAEMON_CLASSPATH"));
```

Comment:
I think it would be better to only support this env variable for daemon process like HistoryServer, ExternalShuffleService and others, like what we did for `SPARK_DAEMON_JAVA_OPTS`. Currently with your fix normal Spark application will also honor this env variable.

Reply:
I agree with you on this. I have made the necessary changes.